### PR TITLE
PUD-1072: Add new endpoint to return list of recall documents for a category

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt [hunk 1]
```diff
@@ -75,6 +76,26 @@ class DocumentController(
...
+    return ResponseEntity.ok(
+      documentService.getAllDocumentsByCategory(recallId, category).map { document ->
+        Api.RecallDocument(
+          document.id(),
+          document.category,
+          document.fileName,
+          document.version,
+          document.details,
+          document.createdDateTime,
+          userDetailsService.get(document.createdByUserId()).fullName()
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt:93]: Future: I can see us wanting the userId on this response as well as the name.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt [hunk 2]
```diff
@@ -111,7 +132,12 @@ class DocumentController(
   @ApiResponses(
     ApiResponse(
       code = 400, message = "Bad request, e.g. virus scan returns error", response = ErrorResponse::class,
-      examples = Example(ExampleProperty(mediaType = "application/json", value = "{\n\"status\": 400,\n\"message\":\"VirusFoundException\"\n}"))
+      examples = Example(
+        ExampleProperty(
+          mediaType = "application/json",
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt:137]: Future: I'd rather we used an imported constant everywhere for this and not a String literal, `"application/json"`.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/AuthenticatedClient.kt [hunk 3]
```diff
@@ -60,6 +62,19 @@ class AuthenticatedClient(
...
 
+  fun getRecallDocuments(recallId: RecallId, category: DocumentCategory): List<Api.RecallDocument> =
+    webTestClient.get().uri { uriBuilder ->
+      uriBuilder.path("/recalls/$recallId/documents")
+        .queryParam("category", "{category}")
+        .build(category)
+    }
+      .headers(addHeaders)
+      .exchange()
+      .expectStatus().isOk()
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/AuthenticatedClient.kt:73]: Alt-Enter on `isOk()` makes a minor suggestion.

**Review by AndyNewton — APPROVED** [review]
Good to go; future comments.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt (hunk 1 above)
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt:93]: Personally, feels like we should avoid exposing uuids to clients. but 🤷

**Pull request merged**